### PR TITLE
Adding max and min PTS to stat report

### PR DESCRIPTION
**Pull request by mosestricks**
small change to state screen
adding the min and the max PTS, this change was made in order to analyze the difference between the last PTS + frame duration to the full duration.

### CODE REVIEW

File: app/src/main/java/com/lightricks/videotricks/media/VideoReader.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
     private CompletableFuture<Void> job;
     private PlaybackControl playbackControl;
     private boolean isDryRun;
+    private long maxPtsObsereved = 0;
```
Comment (bmax-ltx) [inline app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:31]:
```suggestion
    private long maxPtsObserved = 0;
```

File: app/src/main/java/com/lightricks/videotricks/media/VideoReader.java [hunk 2]
```diff
@@ -120,7 +125,9 @@ private void handleInputBuffer(int bufferIndex) {
     private void handleOutputBuffer(int bufferIndex, MediaCodec.BufferInfo bufferInfo) {
         boolean isEOS = (bufferInfo.flags & MediaCodec.BUFFER_FLAG_END_OF_STREAM) != 0;
         boolean isEmpty = bufferInfo.size == 0;
-
+        if (bufferInfo.presentationTimeUs > maxPtsObsereved) {
```
Comment (bmax-ltx) [inline app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:128]: Maybe, this should be done by `StatsCollector` ?

File: app/src/main/java/com/lightricks/videotricks/stats/StatsCollector.java [hunk 3]
```diff
@@ -20,20 +21,10 @@ int getSamplesCount() {
         return inputSamples.size();
     }
 
-    long getActualDuration() {
-        OptionalLong minTime = inputSamples.stream()
+    LongSummaryStatistics getPtsStat() {
```
Comment (bmax-ltx) [inline app/src/main/java/com/lightricks/videotricks/stats/StatsCollector.java:24]: Nice, I didn't know about it!

File: app/src/main/java/com/lightricks/videotricks/media/VideoReader.java (hunk 2 above)
Comment (mosestricks) [reply app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:128]: you probably right

Comment (bmax-ltx) [reply app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:128]: Are you going to move it there?

Comment (mosestricks) [reply app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:128]: yes of course

File: app/src/main/java/com/lightricks/videotricks/media/VideoReader.java [hunk 4]
```diff
@@ -79,6 +80,10 @@ public void setDryRun(boolean dryRun) {
                 .thenRun(handlerThread::quit);
     }
 
+    public long getMaxPtsObsereved() {
```
Comment (bmax-ltx) [inline app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:83]:
```suggestion
    public long getMaxPtsObserved() {
```

File: app/src/main/java/com/lightricks/videotricks/media/VideoReader.java [hunk 5]
```diff
@@ -79,6 +80,10 @@ public void setDryRun(boolean dryRun) {
...
+        return maxPtsObsereved;
```
Comment (bmax-ltx) [inline app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:84]:
```suggestion
        return maxPtsObserved;
```

File: app/src/main/java/com/lightricks/videotricks/media/VideoReader.java (hunk 2 above)
Comment (bmax-ltx) [inline app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:128]:
```suggestion
        if (bufferInfo.presentationTimeUs > maxPtsObserved) {
```

File: app/src/main/java/com/lightricks/videotricks/media/VideoReader.java [hunk 6]
```diff
@@ -120,7 +125,9 @@ private void handleInputBuffer(int bufferIndex) {
     private void handleOutputBuffer(int bufferIndex, MediaCodec.BufferInfo bufferInfo) {
         boolean isEOS = (bufferInfo.flags & MediaCodec.BUFFER_FLAG_END_OF_STREAM) != 0;
         boolean isEmpty = bufferInfo.size == 0;
-
+        if (bufferInfo.presentationTimeUs > maxPtsObsereved) {
+            maxPtsObsereved = bufferInfo.presentationTimeUs;
```
Comment (bmax-ltx) [inline app/src/main/java/com/lightricks/videotricks/media/VideoReader.java:129]:
```suggestion
            maxPtsObserved = bufferInfo.presentationTimeUs;
```

**Comment by mosestricks** [comment]
so you the only one that can merge this.
when you ready...

**Pull request merged**